db/migrations/25-03-07-compute-contribution-stats.sql

```diff
@@ -0,0 +1,133 @@
+                and changed_phrase.state = 'UNAPPROVED'
+                and coalesce(prev_phrase.state, 'UNAPPROVED') = 'UNAPPROVED'
+        ), 0) as words_unapproved_edited
+    from (
+        select
+            DISTINCT ON (ph.language_id, phw.word_id)
+            ph.language_id, phw.word_id,
+            log.*
+        from (
+            (
+                select phrase_id, updated_by, updated_at, gloss, state
+                from gloss
+            ) union all (
+                select phrase_id, updated_by, updated_at, gloss, state
+                from gloss_history
+            ) union all (
+                select
+                    id as phrase_id,
+                    deleted_by as updated_by,
+                    deleted_at as updated_at,
+                    '' as gloss,
+                    'UNAPPROVED' as state
+                from phrase
+                where deleted_by is not null and deleted_at is not null
+            )
+        ) log
+        join phrase_word phw on phw.phrase_id = log.phrase_id
+        join phrase ph on ph.id = log.phrase_id
+        where log.updated_at > date_bin('7 days', date_trunc('day', $1), timestamp '2024-12-15')
+            and log.updated_at < date_bin('7 days', date_trunc('day', $1 + interval '7 days'), timestamp '2024-12-15')
+        order by ph.language_id, phw.word_id, log.updated_at desc
+        ) changed_phrase
+    left join lateral (
+        select h.* from (
+            (
+                select phrase_id, updated_by, updated_at, gloss, state
+                from gloss
+                where exists (
+                        select 1 from phrase_word phw
+                        join phrase ph on ph.id = phw.phrase_id
+                        where ph.id = gloss.phrase_id
+                            and ph.language_id = changed_phrase.language_id
+                            and phw.word_id = changed_phrase.word_id
+                    )
+                    and gloss.updated_at < date_bin('7 days', date_trunc('day', $1 - interval '7 days'), timestamp '2024-12-15')
+            ) union all (
+                select
+                    id as phrase_id,
+                    deleted_by as updated_by,
+                    deleted_at as updated_at,
+                    '' as gloss,
+                    'UNAPPROVED' as state
+                from phrase
+                where deleted_by is not null and deleted_at is not null
+                    and phrase.language_id = changed_phrase.language_id
+                    and exists (
+                        select 1 from phrase_word phw
+                        where phw.phrase_id = phrase.id
+                            and phw.word_id = changed_phrase.word_id
+                    )
+                    and phrase.deleted_at < date_bin('7 days', date_trunc('day', $1 - interval '7 days'), timestamp '2024-12-15')
+            )
+        ) h
+        order by h.updated_at desc
+        limit 1
+    ) prev_phrase on true
```

Comment:
This chunk finds the latest action from the previous week. There's two sources that are separated for index performance reasons and then unioned together to find the latest action:
1. the list of actions taken on glosses for the word
2. the list of phrases with the word that have been deleted